tests/moduletool/test_install.py

```diff
@@ -941,3 +942,386 @@ def test_install_project_with_install_mode_master(tmpdir: py.path.local, snippet
+
+    expected_logs = [
+        ("Installing module minimalv2module (v2)", logging.DEBUG),
+        ("Successfully installed module minimalv2module (v2) version 1.2.3", logging.DEBUG),
+        ("Installing module std (v1)", logging.DEBUG),
+        (
+            """Successfully installed module std (v1) version 3.0.0 in %s from %s"""
+            % (os.path.join(project.downloadpath, "std"), "https://github.com/inmanta/std"),
+            logging.DEBUG,
+        ),
+    ]
+
+    for message, level in expected_logs:
+        log_contains(
+            caplog,
+            "inmanta.module",
+            level,
+            message,
+        )
+
+
+def test_real_time_logging(caplog):
+    """
+    Make sure the logging in _run_command_and_stream_output happens in real time
+    """
+    caplog.set_level(logging.DEBUG)
+
+    cmd: List[str] = ["sh -c 'echo one && sleep 1 && echo two'"]
+    return_code: int
+    output: List[str]
+    return_code, output = PythonEnvironment._run_command_and_stream_output(cmd, shell=True)
+    assert return_code == 0
+
+    assert "one" in caplog.records[0].message
+    assert "one" in output[0]
+    first_log_line_time: datetime = datetime.fromtimestamp(caplog.records[0].created)
+
+    assert "two" in caplog.records[-1].message
+    assert "two" in output[-1]
+    last_log_line_time: datetime = datetime.fromtimestamp(caplog.records[-1].created)
+
+    # "two" should be logged at least one second after "one"
+    delta: float = (last_log_line_time - first_log_line_time).total_seconds()
+    assert delta >= 1
+
+
+def test_pip_output(local_module_package_index: str, snippetcompiler_clean, caplog, modules_v2_dir, tmpdir):
+    """
+    This test checks that pip's output is correctly logged on module install.
+    """
+    caplog.set_level(logging.DEBUG)
+
+    snippetcompiler_clean.setup_for_snippet("")
+
+    modone: module.ModuleV2Metadata = module_from_template(
+        os.path.join(modules_v2_dir, "minimalv2module"),
+        os.path.join(str(tmpdir), "modone"),
+        new_version=version.Version("3.1.2"),
+        new_name="modone",
+        install=True,
+    )
+    modtwo: module.ModuleV2Metadata = module_from_template(
+        os.path.join(modules_v2_dir, "minimalv2module"),
+        os.path.join(str(tmpdir), "modtwo"),
+        new_version=version.Version("2.2.2"),
+        new_name="modtwo",
+        new_requirements=[InmantaModuleRequirement.parse("modone")],
+        install=True,
+    )
+
+    modules = ["modone", "modtwo"]
+    v2_requirements = [Requirement.parse(module.ModuleV2Source.get_package_name_for(mod)) for mod in modules]
+
+    snippetcompiler_clean.setup_for_snippet(
+        f"""
+        import {module.ModuleV2.get_name_from_metadata(modone)}
+        import {module.ModuleV2.get_name_from_metadata(modtwo)}
+        """,
+        autostd=False,
+        python_package_sources=[local_module_package_index],
+        python_requires=v2_requirements,
+        install_project=False,
```

Comment:
Why set `install_project=False` here and `install=True` on the templates?

Reply:
I would suspect that this test will still pass if you remove this whole `setup_for_snippet` as the module_from_template will do an install already

Reply:
Changing this test to `install_project=True` and `install=False` on the templates for consistency